packages/plugins/rudder_plugin_android/android/src/main/java/com/rudderstack/sdk/flutter/managers/UserSessionManager.java

```diff
@@ -0,0 +1,160 @@
+package com.rudderstack.sdk.flutter.managers;
+
+import androidx.annotation.Nullable;
+
+import com.rudderstack.android.sdk.core.RudderClient;
+import com.rudderstack.android.sdk.core.RudderLogger;
+import com.rudderstack.android.sdk.core.util.Utils;
+
+/**
+ * This class is used to manage the user session using the native manual session tracking API i.e., startSession and endSession.
+ */
+public class UserSessionManager {
+  private final long sessionTimeOut;
+  private final PreferenceManager preferenceManager;
+  private final boolean isAutoSessionTrackingEnabled;
+
+  public UserSessionManager(
+    boolean autoSessionTrackingEnabled,
+    boolean autoTrackingLifeCycleEventsEnabled,
+    PreferenceManager preferenceManager,
+    long sessionTimeout) {
+    this.sessionTimeOut = sessionTimeout;
+    this.preferenceManager = preferenceManager;
+    this.isAutoSessionTrackingEnabled = autoSessionTrackingEnabled && autoTrackingLifeCycleEventsEnabled;
+  }
+
+  /**
+   * This method handles the auto session:
+   * <p> If auto session tracking is </p>
+   * <ol>
+   *   <li> enabled in the config, then start a new auto session. </li>
+   *   <li> disabled in the config, then check if there is any existing auto session and end it.</li>
+   * </ol>
+   */
+  public void handleAutoSessionTracking() {
+    if (isAutoSessionTrackingEnabled) {
+      startAutoSessionIfNeeded();
+    } else {
+      endAutoSessionIfExists();
+    }
+  }
+
+  /**
+   * This method starts the auto session:
+   * If there was an existing auto session from the previous run of the app, check if it is expired or not. If it is expired, start a new auto session, else continue with the existing auto session.
+   */
+  public void startAutoSessionIfNeeded() {
```

Comment:
Consider making it private, if this is not called from outside of this class.

Reply:
updated it